sdk/cosmos/azure-cosmos/azure/cosmos/aio/_cosmos_client.py

```diff
@@ -100,6 +99,30 @@ class CosmosClient(object):  # pylint: disable=client-accepts-api-version-keywor
     :type credential: Union[str, Dict[str, str], ~azure.core.credentials_async.AsyncTokenCredential]
     :keyword str consistency_level: Consistency level to use for the session. Default value is None (account-level).
         More on consistency levels and possible values: https://aka.ms/cosmos-consistency-levels
+    :keyword int timeout: An absolute timeout in seconds, for the combined HTTP request and response processing.
+    :keyword int request_timeout: The HTTP request timeout in milliseconds.
+    :keyword str connection_mode: The connection mode for the client - currently only supports 'Gateway'.
+    :keyword proxy_config: Connection proxy configuration.
+    :paramtype proxy_config: ~azure.cosmos.ProxyConfiguration
+    :keyword ssl_config: Connection SSL configuration.
+    :paramtype ssl_config: ~azure.cosmos.SSLConfiguration
+    :keyword bool connection_verify: Whether to verify the connection, default value is True.
+    :keyword str connection_cert: An alternative certificate to verify the connection.
+    :keyword int retry_total: Maximum retry attempts.
+    :keyword int retry_backoff_max: Maximum retry wait time in seconds.
+    :keyword int retry_fixed_interval: Fixed retry interval in milliseconds.
+    :keyword int retry_read: Maximum number of socket read retry attempts.
+    :keyword int retry_connect: Maximum number of connection error retry attempts.
+    :keyword int retry_status: Maximum number of retry attempts on error status codes.
+    :keyword list[int] retry_on_status_codes: A list of specific status codes to retry on.
+    :keyword float retry_backoff_factor: Factor to calculate wait time between retry attempts.
+    :keyword bool enable_endpoint_discovery: Enable endpoint discovery for
+        geo-replicated database accounts. (Default: True)
+    :keyword list[str] preferred_locations: The preferred locations for geo-replicated database accounts.
+    :keyword bool enable_diagnostics_logging: Enable the CosmosHttpLogging policy.
+        Must be used along with a logger to work.
+    :keyword ~logging.Logger logger: Logger to be used for collecting request diagnostics. Can be passed in at client
+        level (to log all requests) or at a single request level.
```

Comment:
Could be worth adding that requests will be logged at `INFO` level (and thus the logger passed in will need to be configured for this)